### PR TITLE
Make Integer#times spec-compliant

### PR DESCRIPTION
**Pull request by richardboehme**
Please take a close look at the compiler change as I'm not sure if this is correct. The process for this fix was more of a "trial and error"-thing than understanding what's going on.

[#201]

### CODE REVIEW

File: test/natalie/integer_test.rb [hunk 1]
```diff
@@ -184,6 +184,24 @@
     end
   end
 
+  describe '#times' do
+    fit 'handles bignums correctly' do
```
Comment (seven1m) [inline test/natalie/integer_test.rb:188]: focused test

File: test/natalie/break_test.rb [hunk 2]
```diff
@@ -176,4 +176,16 @@ def break_in_begin_in_while
     result = break_in_begin_in_while { |x| break :ok }
     result.should == :ok
   end
+
+  fit 'should not leak break flag' do
```
Comment (seven1m) [inline test/natalie/break_test.rb:180]: focused test :joy:

File: lib/natalie/compiler/pass1b.rb [hunk 3]
```diff
@@ -27,12 +27,13 @@ def process_break(exp)
           break_context[:raises_local_jump_error] = true
           exp.new(:raise_local_jump_error, :env, process(value), s(:l, 'LocalJumpErrorType::Break'))
 
-        when :loop, :rescue_do
+        when :loop
+          exp.new(:block, s(:set, var_name, process(value)), s(:c_break))
+        when :rescue_do
```
Comment (seven1m) [inline lib/natalie/compiler/pass1b.rb:32]: This is great! :1st_place_medal:

File: test/natalie/break_test.rb (hunk 2 above)
Comment (richardboehme) [reply test/natalie/break_test.rb:180]: Well... Seems like my self-protection measure did not work 😅 Will fix those tomorrow!

Comment (richardboehme) [reply test/natalie/break_test.rb:180]: I think we need to pass the CI env variable to the docker container stuff somehow to make the "focus"-detection work?

Comment (seven1m) [reply test/natalie/break_test.rb:180]: Yep! I fixed that in 81027a4bf73755256078de21556627678ef7034b just now. :hugs:

**Pull request merged**